LLM Testing/Code-Testing/src/main/java/ChatGPTAPI.java

```diff
@@ -51,7 +55,144 @@ public static void prompts() {
+            promptWriter = new PromptWriter(promptTemplate, responseContent, "t1");
+        }
+        else if (mode.equals("1c")){
+            request = "src/main/java/Task1CompilationFailureRequest.json";
+            promptTemplate = app.getFileFromResource("Prompt Templates/Task1_CompilationError.txt");
+            error = app.getFileFromResource("errors.txt");
+            promptWriter = new PromptWriter(promptTemplate, responseContent, error, "c");
+        }
+        else if (mode.equals("1f")){
+            request = "src/main/java/Task1FailedTestsRequest.json";
+            promptTemplate = app.getFileFromResource("Prompt Templates/Task1_FailedTests.txt");
+            failure = app.getFileFromResource("provided_failures.txt");
+            promptWriter = new PromptWriter(promptTemplate, responseContent, failure, "f");
+        }
+        else if (mode.equals("2")){
+            request = "src/main/java/InitialTask2Request.json";
+            promptTemplate = app.getFileFromResource("Prompt Templates/Task2.txt");
+            promptWriter = new PromptWriter(promptTemplate, responseContent, "t2");
+        }
+        else if (mode.equals("2c")){
+            request = "src/main/java/Task2CompilationFailureRequest.json";
+            promptTemplate = app.getFileFromResource("Prompt Templates/Task2_CompilationError.txt");
+            error = app.getFileFromResource("errors.txt");
+            promptWriter = new PromptWriter(promptTemplate, responseContent, error, "c");
+        }
+        else if (mode.equals("2f")){
+            request = "src/main/java/Task2FailedTestsRequest.json";
+            promptTemplate = app.getFileFromResource("Prompt Templates/Task2_FailedTests.txt");
+            failure = app.getFileFromResource("provided_failures.txt");
+            promptWriter = new PromptWriter(promptTemplate, responseContent, failure, "f");
+        }
+        else {
+            throw new RuntimeException("The prompt argument is invalid.");
+        }
+        String newPrompt = promptWriter.output(); // new prompt in string form
+        Request newRequest;
+        if (args[1] == null){ //default 0.7 temperature
+            newRequest = new Request("gpt-3.5-turbo-16k", newPrompt, 0.7); // object for gson to convert
+        }
+        else{
+            newRequest = new Request("gpt-3.5-turbo-16k", newPrompt, Double.valueOf(args[1])); // object for gson to convert
+        }
+        Gson gson = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
+        JsonElement jsonElement = gson.toJsonTree(newRequest);
+        String jsonString = gson.toJson(jsonElement);
+        try(PrintWriter out = new PrintWriter(request)){
+            out.println(jsonString);
+        }
+        try(PrintWriter out = new PrintWriter("src/main/resources/newPrompt.txt")){
+            out.println(newPrompt);
+        }
+        //sendRequest(request, args[0]);
```

Comment:
Should this be uncommented?

Reply:
Leave it uncommented so that we dont unintentionally call GPT 10 times in a row